From e4f0340d128f719e31bc22786e93cec17f40566b Mon Sep 17 00:00:00 2001
From: root <root@liteserver-8564-1.novalocal>
Date: Mon, 13 Jul 2026 22:32:43 +0800
Subject: [PATCH] huaweicloud: set endpoint route source to cilium host

---
 pkg/datapath/loader/loader.go | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pkg/datapath/loader/loader.go b/pkg/datapath/loader/loader.go
index 0f7b2b4a9..ececaae4c 100644
--- a/pkg/datapath/loader/loader.go
+++ b/pkg/datapath/loader/loader.go
@@ -94,6 +94,13 @@ func upsertEndpointRoute(ep datapath.Endpoint, ip net.IPNet) error {
 		Device: ep.InterfaceName(),
 		Scope:  netlink.SCOPE_LINK,
 	}
+	if option.Config.IPAM == ipamOption.IPAMHuaweiCloud {
+		if ip.IP.To4() != nil {
+			endpointRoute.Local = node.GetInternalIPv4Router()
+		} else {
+			endpointRoute.Local = node.GetIPv6Router()
+		}
+	}
 
 	return route.Upsert(endpointRoute)
 }
-- 
2.33.0

